exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/PrometheusCollectorBuilder.java

```diff
@@ -8,10 +8,11 @@
 import io.opentelemetry.sdk.metrics.export.MetricProducer;
 import io.prometheus.client.Collector;
 import java.util.Objects;
+import javax.annotation.Nullable;
 
 /** Builder for {@link PrometheusCollector}. */
 public class PrometheusCollectorBuilder {
-  private MetricProducer metricProducer;
+  @Nullable private MetricProducer metricProducer;
```

Comment:
I would normally change to having the builder constructor accept this, but I know this whole class is under the scalpal in #3578

Reply:
Speaking of which, would you be ok removing conflicting parts from this PR and moving into that PR?

Reply:
I'll merge main into your PR if this gets merged first, it will be helpful to get the artifacts working with nullaway if possible